tests/framework-tests/core/js/ModelTransformationTests.js

```diff
@@ -2652,10 +2750,274 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+                outputPath: "element",
+                inputPath: "value"
+            }]
+        },
+        method: "assertDeepEq",
+        model: {
+            element: 1
+        },
+        expected: {
+            value: "dog"
+        },
+        fullyinvertible: true
+    }, {
+        message: "dereference() should ignore notFound when the value is found in the array.",
+        transform: {
+            value: {
+                transform: {
+                    type: "fluid.transforms.dereference",
+                    array: ["sheep", "dog"],
+                    notFound: "notFound",
+                    inputPath: "element"
+                }
+            }
+        },
+        invertedRules: {
+            transform: [{
+                type: "fluid.transforms.indexOf",
+                array: ["sheep", "dog"],
+                notFound: "notFound",
+                outputPath: "element",
+                inputPath: "value"
+            }]
+        },
+        method: "assertDeepEq",
+        model: {
+            element: 0
+        },
+        expected: {
+            value: "sheep"
+        },
+        fullyinvertible: true
+    }];
+
+    fluid.tests.transforms.dereferenceBoundaryTests = [{
+        message: "dereference() should return undefined when the given index is -1.",
+        transform: {
+            type: "fluid.transforms.dereference",
+            array: ["sheep", "dog"],
+            value: -1
+        },
+        expected: undefined
+    }, {
+        message: "dereference() should return undefined when the \"array\" is arrayable but the given index is -1.",
+        transform: {
+            type: "fluid.transforms.dereference",
+            array: "dog",
+            value: -1
+        },
+        expected: undefined
+    }, {
+        message: "dereference() should return undefined when the index is not provided.",
+        transform: {
+            type: "fluid.transforms.dereference",
+            array: ["sheep", "dog"]
+        },
+        expected: undefined
+    }, {
+        message: "dereference() should return undefined when the calculated index is -1.",
+        transform: {
+            type: "fluid.transforms.dereference",
+            array: ["sheep", "dog"],
+            value: 2,
+            offset: 3
+        },
+        expected: undefined
+    }, {
+        message: "dereference() should return what's defined in the notFound when the value is not found in the array.",
+        transform: {
+            type: "fluid.transforms.dereference",
+            array: ["sheep", "dog"],
+            value: -1,
+            notFound: "notFound"
+        },
+        expected: "notFound"
+    }];
+
+    jqUnit.test("fluid.transforms.dereference()", function () {
+        fluid.tests.transforms.testOneStructure(fluid.tests.transforms.deferenceTests);
+
+        fluid.tests.transforms.testOneStructure(fluid.tests.transforms.dereferenceBoundaryTests, {
             transformWrap: true,
```

Comment:
To these tests you should add a couple of "round trip" inversion tests, using the testOneInversion driver at https://github.com/fluid-project/infusion/blob/master/tests/framework-tests/core/js/ModelTransformationTests.js#L71 - these should include at least one test which features "offset".